### DISCUSSION TITLE
Week 4: Artificial Intelligence

### DISCUSSION THREAD

**Discussion by eculler** [earthlab-education/Earth-Analytics-2023-01-Intro#301] [Discussion submissions]
AI is too broad a topic for a 30 minute discussion! But, we will attempt to cover some key topics relevant to your work as Earth Data Scientists.

The readings for this week are:

1. As a complement to the exercises you are completing in writing code using ChatGPT: [How to use ChatGPT: Data Analyst and Data Scientist Use Cases](https://medium.com/learning-data/how-to-use-chatgpt-data-analyst-data-scientist-use-cases-f8055bbdfdfc) from John Pauler on Medium [Pauler, J. (2023, June 28). How To Use ChatGPT: Data Analyst & Data Scientist Use Cases. Learning Data. https://medium.com/learning-data/how-to-use-chatgpt-data-analyst-data-scientist-use-cases-f8055bbdfdfc]
2. July of this year was a great month for those who, like me, enjoy discussing (and, let's be honest - making fun of) image classifiers. Why? The birth of a giraffe without spots! Wildlife cameras are one of the potentially extremely useful sources of Big Data in EDS, if only we could get them to work the way we want. Read more in [AI vs. a giraffe with no spots](https://www.aiweirdness.com/ai-vs-a-giraffe-with-no-spots/) from Janelle Shane's AI Weirdness blog [Shane, J. (2023, August 28). AI vs a giraffe with no spots. AI Weirdness. https://www.aiweirdness.com/ai-vs-a-giraffe-with-no-spots/]
3. This one is a bit long - we may not get to it in class - but it defines a lot of the ways I think about AI and machine learning. I highly recommend it for anyone who wants to work on AI/ML models of physical systems, or use data generated by such models. In [Deep Learning is Hitting a Wall](https://nautil.us/deep-learning-is-hitting-a-wall-238440/), cognitive scientist and AI expert Gary Marcus argues that to truly reap the benefits of AI we need a combined data-driven and symbolic or theoretical approach. An earth science example - Conservation of mass and energy represent a huge advance in our understanding of the Earth System, and yet many machine learning models throw out these incredibly important theoretical concepts in favor of more and more data. 

Some questions to get you started:

- Describe some advantages and drawbacks of coding using large language models like ChatGPT
- From the article or your experience with LLMs, what are some best practices for writing useful prompts for generative AI? What types of questions can be used to confuse or trip up the LLMs?
- Suppose you are assembling data to train and AI model to track the movement of apex predators in the Los Angeles area across the [Wallis Annenberg Wildlife Crossing](https://101wildlifecrossing.org/).
    - What kinds of images would you need to include? 
    - How could you test your dataset and model to uncover potential biases? 
    - What kinds of scientific questions could you ask with such a dataset?
    - How might the question influence your data needs?
    - How might your view of animals (for example as property, resources, entities with rights and legal status, or as family) affect how you approach a study like this?
- Would you trust a 100% AI-based flood early warning system? We're talking here about issuing evacuation notices, not about seasonal forecasting. Most systems in the US today rely at least partially on physically-based atmospheric and/or land surface models.

**Comment by ellenalamont17** [comment]
Reading the first article, "How to Use ChatGPT...", it took me back to early discussions I had with fellow educators at all grade and academic levels. I remember when ChatGPT first came out, educators were scrambling with how to combat the rapid growth of the use of the LLM platform in the  production of class assignments. For many, it felt like AI was overtaking the classroom, robbing students of their ability to actually learn content, and many teachers were in crisis. 

Further were concerns over plagiarism and content theft. Because these systems scrub through content available online (sometimes with questionable protections), the material generated in these LLM platforms often failed to provide proper attribution and in worse cases, lifted entire passages from available resources. It is for similar reasons that workers in government agencies are prohibited from using platforms, such as ChatGPT. Initially, workers thought to use the platform to proofread/review documents or to quickly produce executive summaries of documents to provide to stakeholders. However, it became quickly apparent that such activities could pose security threats, because their use in such systems makes the content from the government documents available for search in others' queries. 

Overtime, it seemed that my educator friends realized that these technologies were not going anywhere, so to speak, and many began to embrace the technology. Most acknowledged improvements to the students efficiency in producing more quality material, the ability for a student to obtain the answers they want to a question more quickly so that they can resume at-hand tasks, and the ability to promote deeper exploration into a topic or theme. 

I would say that all of these individuals, however, note the critical need to educate their students on the benefits and harms of the use of such platforms. They also have placed more emphasis on communication and thinking skills while incorporating these technologies. For example, one educator in English literature requires their students to write their own drafts of a report, even if using an LLM for content research, before submitting it for review and improvement by AI. The student is then required to submit both the original and updated documents along with a comparative analysis between the two versions that outlines some of the ways they could improve their own normal writing based on the AI suggestions. 

The key is realizing that AI is a tool, not a solution. Given that AI is "the future," it is somewhat of a disservice to keep people away from these tools when their can truly help improve upon the work already being done. And as suggested in the article, "While AI likely won’t take your job, another Analyst who uses AI might". What keeps humans ahead of AI is our ability to think critically, check all information for accuracy, and employ our own working experiences and judgement.

**Reply by eculler** [reply]
I 100% agree about LLMs being a tool! If I was going to pick something to use an LLM for, though I don't think it would be content research. They just aren't consistently accurate - they're much better at creating the illusion of accuracy. For example, they do things like cite non-existent sources to make it seem like their information is founded.

**Comment by ellenalamont17** [comment]
One thing that struck me while reading the "Deep Learning is Hitting a Wall" article lies in the very definition of deep learning - a technique for recognizing patterns. With natural systems, the use of AI seems really tricky, because not only do these systems have to recognize patterns, but also systematic and non-systematic changes in patterns through time. While at large scales natural processes, environments, and visuals can be generally identified or understood, but in most cases, where researchers are concerned, we are often operating on the fringe of natural systems where things so not necessarily follow regularly identified patterns or stray from the norm in some way. 

It made me think about the struggle I had several years back with trying to develop a relative age dating index for alluvial fans in desert environments. With time, alluvial fans in deserts develop "varnish" on the fan surface and "pavement" from the progressive compaction, blowing, and reworking of materials on the fan surface. With the development of fans of different ages, the variation in textures should be readily identifiable by an AI, yet a suitable identification system has yet to be developed, because every fan has other varying unique features that throw off the identification algorithm. (e.g. different color materials that can be mistaken as dark varnish, coarser or finer materials composing the fan, and larger or smaller bar-swale topography that skews the surface roughness). It seems really hard then to use deep learning for natural systems. What are better ways to address such concerns? I'd be curious what it would take...as I am sure many developers are as well.

**Reply by eculler** [reply]
Super-interesting application!

I think what a lot of folks are talking about is combining symbolic or physically-based modeling with deep learning (e.g. physics-aware AI). In the case of alluvial fans - perhaps the images could be enhanced in pre-processing (e.g. through a spectral index) in such a way as to bring out the age-related textural differences while suppressing morphological differences. In that way, you are directing the AI to the important part of the image. 

In many cases, the challenge for natural systems is at least partially related to the lack of data - there are lots of images of natural systems, but they are not necessarily labeled. Labeling and selecting data for AI can be thought of as programming the model - it's a really crucial part of the process that often gets ignored. Sometimes, you can expand a dataset artificially by transforming images, e.g. flipping them, which might be a useful strategy for alluvial fans.

**Comment by Adriannaluger** [comment]
The first article really surprised my that chatGPT was the lasted growing product of all time. But, at the same time it makes so much sense and seems so reasonable for how fast it really took over the world. I think personally I can see myself using chatGPT to improve my efficiency and to check my work but there are many more benefits to the platform. I think for people in the technology world they are constantly being asked to evolve their knowledge so it makes sense that AI fits into this criteria. Some tips when using AI is to be clear/specific, provide context, establish roles, and set the tone.  The giraffe with no spots is so interesting, and a great example of the imperfections of AI. AI does best with images it has seen before (which in this case it never was), and it gives you the most likely answer and not necessarily the most correct answer. On deep learning the thought of AI taking over radiologists sounds pretty terrifying. I definitely would want a human to be checking my scans personally. I guess I also just wonder why we would want this? I understand it can make us more productive as a society but in general it seems more like its just going to put millions of people out of jobs and at that point when does it end, sort of seems like an endless loop of AI taking over everything? For example, AI could take over professors but to me you just loose so much when switching to technology instead of human interactions from my perspective, that when is the line going to be drawn?

**Reply by eculler** [reply]
Some great points!

I personally think it's unlikely that AI will totally take over industries like radiology or journalism. Like the article says, right now it looks like AI won't be taking over those jobs - but people who use AI might. But many of the hazards of AI are already with us, namely that they perpetuate and amplify existing human biases. Take for example resume-scanning software. Several programs had to be dropped because they were so biased against women and racial minorities, using subtle clues like which schools people attended and sports teams to figure out details about people. AIs are really good at finding shortcuts, whether or not those shortcuts are moral or just.

**Reply by annettebalmes** [reply]
I like to think that research scientists are safe for quite awhile as well. Can't replicate work that doesn't exist just yet!

**Comment by jsmi374** [comment]
I am new to the ChatGPT world, and this is my first introduction to the platform. It sounds like a useful tool, that can help researchers, data analysist, code developers, and students increase efficiency, performance, and success rates in applying complex workflows across a wide range of software suites and coding languages. Before using any tool it is best to know and understand its limitations .... which in this case appears to be a moving target, as stated in the first article. Critical thinking is paramount! It's okay and encouraged to use this tool as it has many benefits, but users need to perform quality checks along the way and test the results prior to integration. It is a little scary to be using a tool that pulls its knowledge base from the open internet, and its inability to know good from bad. I see great value in using this tool to speed up lower level task and increase optimization. Personally, I would be hesitant to let it take over upper level task and inform research practices. Quality is on the line, the answer is not always what is next? but what is best? 

Having said that ... some aspects of ChatGPT are appealing to me. I am excited to learn more about this platform, and how I can incorporate it into my workflows.

**Reply by eculler** [reply]
> the answer is not always what is next? but what is best?

100%!

**Comment by peterkobylarz** [comment]
There was a ton of buzz and anxiety about ChatGPT within my organization earlier in the year when it first blew up. We had several large group calls discussing the implications of LLM AI for our field (Environmental Consulting), where many people expressed concern and/or excitement, including the "are our jobs going to be replaced?" question that I'm sure is familiar to many other people here. Shortly thereafter, the buzz died off and I haven't heard much chatter about ChatGPT at work. I would not be surprised if it re-emerges in the near future, especially with some of the improvements to ChatGPT that have happened since then.

I've tried to use it from time to time to assist with GIS related workflows and run into some frustrating results where my prompts generate workflows that read like real GIS documentation but are mixing in completely made up tools and processes with real ones. After reading through the readings this week, I wonder how much of this is due to poor/lazy prompt creation? In any event, it gave me some early skepticism about relying too heavily on ChatGPT for specific technical output. I have found that these ChatGPT sessions are always good for brainstorming though and I often still feel like I got something valuable out of the process even if I don't actually end up using the raw output from ChatGPT.

I have had more luck using ChatGPT while debugging Python macros and scripts. It does kind of make me feel like I am "cheating" though... There's less of a need to internalize syntax and API tools, but I guess that is not necessarily a bad thing!

**Reply by nbol48** [reply]
I've also run into some serious limitations of ChatGPT, but in discussions with friends, we continue circling back to the nascence of this type of product and the INCREDIBLE amount of money now pouring into the generative AI bucket. There seems to have been hesitance (warranted) on the part of legacy tech corporations, but once ChatGPT stepped across the public access LLM product line, it seems as if everyone is developing their own version. All this to say, the tool is still in its "early adoption" phase, despite the record breaking growth of ChatGPT, and its abilities are only going to increase. 

Ultimately, I think a diversity of products with refined use-cases are quickly coming down the pipe, and as tools become more suited to their intended tasks, the "prompt engineering" burden will be reduced on everyday users both recreational and commercial.

Exciting times!

**Reply by NobodyLivesHere** [reply]
This has gotten a lot better on GPT 4 (which is subscription only unless you are using Bing) but it does have a habit of very authoritatively asserting wrong information. Usually this happens because it sees conflicting information in its sources and associates made-up tokens with the prompt response to remedy this.  So when it spits out wrong info for GIS workflows most likely was also looking at some other unrelated data processing description and trying to work it in to the answer.  

The good news is that if you tell it that it's wrong or ask if to verify itself, it will usually correct itself unless the correct info is just not within its sources.  GPT 4 is handy because it has the ability to read in additional information through the internet and is also really good at pulling up scientific literature on almost anything imaginable. 

But you are right, a lot of it just boils down to prompt engineering.  Telling it to take on a certain role and to avoid doing certain things in its responses goes a long way towards getting it to behave more.  It's somewhat like doing modern day document translation: machine translation has gotten good enough to the point that for most widely spoken languages, it can return accurate translations but sometimes they worded very awkwardly or have some sort of contextual information left out, so you still need someone who actually knows the languages to oversee it in order to get quality work.

**Comment by lucap1211** [comment]
I've just read [this](https://www.nytimes.com/2023/05/01/business/ai-chatbots-hallucination.html?searchResultPosition=59) article and found this quote especially interesting regarding the accuracy and trust aspects of AI. “Counterintuitively, hallucinations can become more dangerous as models become more truthful, as users build trust in the model when it provides truthful information in areas where they have some familiarity,”.  It is already quite challenging to navigate and differentiate false information from legit sources and generating more content can make it even more complicated.

**Reply by annettebalmes** [reply]
I am dealing with something similar in my own research. I am creating a new version of the crustal magnetic field map. Part of that is interpolating data, and a lot of places where I don't have enough data and have to come up with something. As I look to validate my map, I use geologic mapping and tectonic boundaries to help decide if what I did was correct or not. But each of those methods use each other to make their maps. So at what point have we all hallucinated data, and confirm it using each other maps! And that's people doing this, I can't imagine trying to put this through an AI model...

**Comment by JKinsler** [comment]
I really enjoyed the example of the spotted giraffe. 'Obvious' information is different to a human and a computer.

**Comment by annettebalmes** [comment]
The way you perceive animals can significantly shape your approach to a study tracking apex predators in the Los Angeles area via the Wallis Annenberg Wildlife Crossing. Different views of animals profoundly influence the study's goals and methods.

For those seeing animals as property or resources, the study would prioritize management to safeguard human interests, including livestock protection, hunting, or tourism. Conversely, perceiving animals as entities with inherent rights and legal status would emphasize ethical and legal considerations, emphasizing minimal human interference and the well-being of apex predators. Viewing animals as integral to biodiversity and ecosystems would steer the study toward conservation, with a focus on understanding the apex predators' ecological roles and the impact of the Wildlife Crossing on preserving their populations and habitats. For those who see animals as part of an extended ecological family, the study would prioritize well-being and sustainability, striving to minimize human disruption to natural behaviors and migration patterns.

Ultimately, one's ethical and philosophical perspective on animals can guide the study's objectives, methodology, and ethical considerations in conservation and research efforts in the area.

**Comment by thku8507** [comment]
The Nautilus article is very thought-provoking! I didn't know that AI creators were banking on scaling, the idea that AI models would continuously get better with larger training datasets, and that there's now some evidence that beyond a certain point scaling increases fluency but doesn't increase accuracy. The article was published in March 2022, half a year before ChatGPT was released. I wonder if the author's perspective on AI may have changed after using ChatGPT (I know mine did). 

I have a hard time squaring the useful and scary sides of AI. On the one hand, I know from experience that it's undeniably super useful for coding, and I'm sure it's helpful for many other things too. In low stakes applications like learning to code, where you can quickly find out if it's wrong and it doesn't matter either way, I see very few downsides. 

But on the other hand, I have profound mistrust for many of those in charge of developing AI and using it for high-stakes purposes. The idea of this powerful tool's future being shaped by a small group of extremely wealthy software tycoons motivated by profit, ego, competition, and grandiose techno-utopian visions of the future is not pleasant. Take this quote from a [conversation ](https://www.nytimes.com/2023/12/03/technology/ai-openai-musk-page-altman.html) between Elon Musk and Google's CEO at the time Larry Page from 2015, where the latter was arguing for AI's unmitigated development: 

> "With a rasp of frustration, Mr. Page insisted his utopia should be pursued. **Finally he called Mr. Musk a “specieist,” a person who favors humans over the digital life-forms of the future**. 

It's worth noting that Musk has largely changed his tune. So did the founders of OpenAI, who always planned on keeping their company private so they could make sure their technology was developed responsibly, until competition drove them to sell. 

Meanwhile, the US military has rejected efforts to limit research into [drones that can autonomously decide to kill](https://www.businessinsider.com/us-closer-ai-drones-autonomously-decide-kill-humans-artifical-intelligence-2023-11).

**Comment by eculler** [comment]
Eep yeah, these are some pretty scary example! I think part of the issue is that we keep giving fancy names to technology that doesn't live up to it yet. I'm not sure the GPT models can really be called intelligent, but they're certainly very useful.

I've been follow the TBD coverage out of U of Arizona -- it's not super technical, but I think they do a good job of reporting on lots of different potential ramifications of technology.
